src/protocol.ts

```diff
@@ -20,12 +35,10 @@ export class Protocol {
 
   private provider: JsonRpcProvider;
   private positionManager: PositionManager;
-  private raftCollateralToken: ERC20Indexable;
-  private raftDebtToken: ERC20Indexable;
 
-  private _collateralSupply: Decimal | null = null;
-  private _debtSupply: Decimal | null = null;
-  private _borrowingRate: Decimal | null = null;
+  private _collateralSupply: CollateralTotalSupply[] | null = null;
+  private _debtSupply: DebtTotalSupply[] | null = null;
+  private _borrowingRate: BorrowingRate[] | null = null;
```

Comment:
`Record<UnderlyingCollateralToken, Decimal>` is a better interface for these as you have a map + can easily be converted to a list via `Object.entries`.